core/domain/exp_domain.py

```diff
@@ -2190,6 +2229,102 @@ def _convert_states_v27_dict_to_v28_dict(cls, states_dict):
             }
         return states_dict
 
+    @classmethod
+    def _convert_states_v28_dict_to_v29_dict(cls, states_dict):
+        """Converts from version 28 to 29. Version 29 added image assets field
+        in state model.
+
+         Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        image_counter = 0
+        for state_dict in states_dict.itervalues():
+            image_assets = {}
+            image_mapping = {}
+
+            content_html = state_dict['content']['html']
+            (new_content_html, image_counter, image_mapping) = (
+                html_validation_service.get_html_with_image_id_in_image_tag(
```

Comment:
This is still the same, I think we had a discussion on how to handle this properly isn't it?

Reply:
yup, will do in the next commit.